torch/tensortypes/nestedtensor/nested.py

```diff
@@ -0,0 +1,215 @@
+import torch
+
+# This implementation is based on NestedTensor 0.0.1
+# RFC: https://github.com/pytorch/pytorch/issues/22169
+
+def is_nested_tensor(obj):
+    return isinstance(obj, NestedTensor)
+
+
+# Arguments match torch.tensor
+def make_nested_tensor(data, dtype=None, device=None, requires_grad=False, pin_memory=False):
+    if is_nested_tensor(data):
+        # This is consistent with torch.tensor(torch.Tensor)
+        # but errors out.
+        raise ValueError("UserWarning: To copy construct from a NestedTensor, "
+                         "use sourceTensor.clone().detach() or "
+                         "sourceTensor.clone().detach().requires_grad_(True), "
+                         "rather than torch.tensor(sourceTensor).")
+    elif torch.is_tensor(data):
+        # The user has the right to expect a NestedTensor from this
+        # function, but we can't meaningfully provide one if passed a Tensor
+        raise ValueError("Can't construct a NestedTensor from a Tensor")
```

Comment:
What is wrong with just unbinding the tensor?

Reply:
It doesn't adhere to the contract of the constructor. If the user wants this, then she should call torch.nestedtensor(data.unbind())